### PR TITLE
Correct permissions bitmask

### PR DESCRIPTION
**Pull request by megumisonoda**
This sets correct bitmask values and ordering. It also adds the external emoji permission to the hash, enabling `can_use_external_emoji`.

Values were taken from Discord's official API page. For the External Emoji permission value, I created a role which had only that permission, then returned its bitmask. All values were verified myself.

### CODE REVIEW

File: lib/discordrb/permissions.rb [hunk 1]
```diff
@@ -10,7 +10,7 @@ class Permissions
       0 => :create_instant_invite, # 1
       1 => :kick_members,          # 2
       2 => :ban_members,           # 4
-      3 => :manage_roles,          # 8, also Manage Permissions
+      3 => :administrate,          # 8
```
Comment (meew0) [inline lib/discordrb/permissions.rb]: This permission is called "Administrator", not "Administrate", could you change this?

**Comment by megumisonoda** [comment]
Will do.

File: lib/discordrb/permissions.rb (hunk 1 above)
Comment (megumisonoda) [inline lib/discordrb/permissions.rb]: This is because it's going to get wrapped in "can_administrate". "can_administrator" sounds... bad, frankly.

Comment (VxJasonxV) [inline lib/discordrb/permissions.rb]: We discussed this in #ruby_discordrb about 12 hours ago. Picking a grammatically correct name for the sake of the generated method names was my suggestion. `can_administrator?` just sounds silly.

Can you use `alias_method` onto a method that doesn't exist? Alias `:can_administrate?` to `:can_administrator?` despite the fact that it doesn't exist in the first place?

Comment (meew0) [inline lib/discordrb/permissions.rb]: Well, the method exists, just isn't explicitly defined. What I would do is call the bit `administrator`, then add an
```rb
alias_method :can_administrate=, :can_administrator=
```
in Permission, and similarly `can_administrate?`/`can_administrator?` in Member.

**Comment by megumisonoda** [comment]
Should be good to merge into the main repo now. Sorry it took a while; had some IRL stuff to deal with.

File: lib/discordrb/data.rb [hunk 2]
```diff
@@ -247,7 +247,7 @@ def permission?(action, channel = nil)
       # (Coincidentally, Manage Permissions is the same permission as Manage Roles, and a
       # Manage Permissions deny overwrite will override Manage Roles, so we can just check for
       # Manage Roles once and call it a day.)
-      return true if defined_permission?(:manage_roles, channel)
+      return true if defined_permission?(:administrate, channel)
```
Comment (megumisonoda) [inline lib/discordrb/data.rb]: Done.

**Pull request merged**